cmd/cmd/backends.go

```diff
@@ -0,0 +1,149 @@
+package cmd
+
+import (
+	"fmt"
+
+	"github.com/spf13/cobra"
+)
+
+var (
+	bindAddr string
+	// should port be a string?
+	port                   int
+	tlsConfig              string
+	connectionCost         float64
+	nodeCost               map[string]int
+	allowedPeers           []string
+	remoteAddress          string
+	keepRedial             bool
+	tlsClientConfig        string
+	localAddress           string
+	localPort              int
+	udpPeerAddress         string
+	udpPeerRedial          bool
+	udpPeerCost            float64
+	udpPeerAllowedPeers    []string
+	wsListenerBindAddr     string
+	wsListenerPort         int
+	wsListenerPath         string
+	wsListenerTLS          string
+	wsListenerCost         float64
+	wsListenerAllowedPeers []string
+	wsPeerAddress          string
+	wsPeerRedial           bool
+	wsPeerExtraHeader      string
+	wsPeerTLS              string
+	wsPeerCost             float64
+	wsPeerAllowedPeers     []string
+)
+
+var tcpListener = &cobra.Command{
+	Use:   "tcp-listener",
+	Short: "Run a backend listener on a TCP port",
+	Long:  ``,
+	Args:  cobra.NoArgs,
+	Run: func(cmd *cobra.Command, args []string) {
+		fmt.Printf("tcpListener called with port %v\n", port)
+	},
+}
+
+var tcpPeer = &cobra.Command{
+	Use:   "tcp-peer",
+	Short: "Make an outbound backend connection to a TCP peer",
+	Long:  ``,
+	Args:  cobra.NoArgs,
+	Run: func(cmd *cobra.Command, args []string) {
+		fmt.Println("resources called")
+	},
+}
+
+var udpListener = &cobra.Command{
+	Use:   "udp-listener",
+	Short: "Run a backend listener on a UDP port",
+	Long:  ``,
+	Args:  cobra.NoArgs,
+	Run: func(cmd *cobra.Command, args []string) {
+		fmt.Println("resources called")
+	},
+}
+
+var udpPeer = &cobra.Command{
+	Use:   "udp-peer",
+	Short: "Make an outbound backend connection to a UDP peer",
+	Long:  ``,
+	Args:  cobra.NoArgs,
+	Run: func(cmd *cobra.Command, args []string) {
+		fmt.Println("resources called")
+	},
+}
+
+var wsListener = &cobra.Command{
+	Use:   "ws-listener",
+	Short: "Run an http server that accepts websocket connections",
+	Long:  ``,
+	Args:  cobra.NoArgs,
+	Run: func(cmd *cobra.Command, args []string) {
+		fmt.Println("resources called")
+	},
+}
+
+var wsPeer = &cobra.Command{
+	Use:   "ws-peer",
+	Short: "Connect outbound to a websocket peer",
+	Long:  ``,
+	Args:  cobra.NoArgs,
+	Run: func(cmd *cobra.Command, args []string) {
+		fmt.Println("resources called")
+	},
+}
+
+func init() {
+
+	tcpListener.Flags().StringVar(&bindAddr, "bindaddr", "0.0.0.0", "Local address to bind to (default: 0.0.0.0)")
```

Comment:
The original message also included the var type, e.g. `bindaddr=<string>: Local address to bind to (default: 0.0.0.0)` do we want to also include the `=<string>` here :thinking:

Reply:
and by extension all the other flags